consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/ErrorControllerAdvice.scala

```diff
@@ -16,32 +16,44 @@
 
 package za.co.absa.spline.consumer.rest.controller
 
+import org.springframework.beans.TypeMismatchException
 import org.springframework.http.HttpStatus._
 import org.springframework.http.ResponseEntity
+import org.springframework.http.converter.HttpMessageConversionException
 import org.springframework.web.bind.annotation.{ControllerAdvice, ExceptionHandler}
 import org.springframework.web.context.request.async.AsyncRequestTimeoutException
-import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException
 import za.co.absa.spline.common.logging.{ErrorCode, ErrorMsg}
 import za.co.absa.spline.common.webmvc.NonStandardResponseEntity
+import za.co.absa.spline.consumer.rest.controller.ErrorControllerAdvice._
 
 @ControllerAdvice(basePackageClasses = Array(classOf[_package]))
 class ErrorControllerAdvice {
 
   @ExceptionHandler(Array(
     classOf[NoSuchElementException]
   ))
-  def handle_404 = new ResponseEntity(NOT_FOUND)
+  def notFound: ResponseEntity[_] = new ResponseEntity(NOT_FOUND)
 
   @ExceptionHandler(Array(
-    classOf[MethodArgumentTypeMismatchException]
+    classOf[TypeMismatchException],
+    classOf[HttpMessageConversionException]
   ))
-  def handle_400(e: MethodArgumentTypeMismatchException) = new ResponseEntity(ErrorMsg(e.getMessage), BAD_REQUEST)
+  def badRequest(e: Exception): ResponseEntity[_] = new ResponseEntity(ErrorMsg(e.getMessage), BAD_REQUEST)
 
   @ExceptionHandler(Array(
     classOf[AsyncRequestTimeoutException]
   ))
-  def handle_598(e: AsyncRequestTimeoutException) = NonStandardResponseEntity(598, ErrorCode(e))
+  def asyncTimeout(e: Exception): ResponseEntity[_] = NonStandardResponseEntity(ASYNC_TIMEOUT, ErrorCode(e))
 
   @ExceptionHandler
-  def handle_500(e: Throwable) = new ResponseEntity(ErrorCode(e), INTERNAL_SERVER_ERROR)
+  def serverError(e: Throwable): ResponseEntity[_] = new ResponseEntity(ErrorCode(e), INTERNAL_SERVER_ERROR)
+}
+
+object ErrorControllerAdvice {
+  /**
+    * This could probably be replaced by the standard 503 response code.
+    * See: https://github.com/AbsaOSS/spline/issues/474
+    */
+  private val ASYNC_TIMEOUT = 598
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field name 'ASYNC_TIMEOUT' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4536902)